docs/en/testing/supplemental.md

```diff
@@ -1,378 +1,454 @@
 
-## Harness API
+Working with [assertion templates](/missing/link) and the test renderer is done using [wrapped test nodes](/missing/link) that are defined in the assertion templates structure, ensuring type safety throughout the testing life-cycle.
 
-```ts
-interface HarnessOptions {
-	customComparators?: CustomComparator[];
-	middleware?: [MiddlewareResultFactory<any, any, any>, MiddlewareResultFactory<any, any, any>][];
-}
+The expected structure of a widget is defined using an assertionTemplate and passed to the test renderer's [`.expect()`](/missing/link) function which executes the assertion.
+
+> src/MyWidget.spec.tsx
+
+```tsx
+import { tsx } from '@dojo/framework/core/vdom';
+import renderer, { assertionTemplate } from '@dojo/framework/testing/renderer';
+
+import MyWidget from './MyWidget';
+
+const baseTemplate = assertionTemplate(() => (
+	<div>
+		<h1>Heading</h1>
+		<h2>Sub Heading</h2>
+		<div>Content</div>
+	</div>
+));
+
+const r = renderer(() => <MyWidget />);
+
+r.expect(baseTemplate);
+```
 
-harness(renderFunction: () => WNode, customComparators?: CustomComparator[]): Harness;
-harness(renderFunction: () => WNode, options?: HarnessOptions): Harness;
+## Wrapped Test Nodes
+
+In order for the test renderer and assertion templates to be able to identify nodes within the expected and actual node structure a special wrapping node must be used. The wrapped nodes can get used in place of the real node in the expected assertion template structure, maintaining all the correct property and children typings.
+
+To create a wrapped test node use the `wrap` function from `@dojo/framework/testing/renderer`:
+
+> src/MyWidget.spec.tsx
+
+```tsx
+import { wrap } from '@dojo/framework/testing/renderer';
+
+import MyWidget from './MyWidget';
+
+// Create a wrapped node for a widget
+const WrappedMyWidget = wrap(MyWidget);
+
+// Create a wrapped node for a vnode
+const WrappedDiv = wrap('div');
 ```
 
--   `renderFunction`: A function that returns a `WNode` for the widget under test
--   [`customComparators`](/learn/testing/dojo-test-harness#custom-comparators): Array of custom comparator descriptors. Each provides a comparator function to be used during the comparison for `properties` located using a `selector` and `property` name
--   `options`: Expanded options for the harness which includes `customComparators` and an array of middleware/mocks tuples.
+The test renderer uses the location of a wrapped test node in the expected tree to attempt to perform the requested action (either `r.property()` or `r.child()`) on the actual output of the widget under test. If the wrapped test node does not match the corresponding node in the actual output tree then no action will be performed and the assertion will report a failure.
```

Comment:
It seems like this statement is made out of context here

Reply:
@maier49 It's trying to explain how the wrapped nodes are used, is there a more appropriate place outside of the test node section?

Reply:
Updated